### PR TITLE
Fix gradient recording in Quickstart

### LINKED ISSUE #188: Force calculation yields different results using different methods
After using PMECalculator to find the potential, I wish to find the forces. I tried the code below. 

```
energy_PME = charges.T @ potential_PME

#The code below is used in the example in README 
energy_PME.backward()
forces_PME = -positions.grad

#This code also used to find force in torchpme  
forces_autograd = torch.autograd.grad(-energy_PME, positions)[0]
```

Results from forces_autograd is the same as the forces stated in coulomb_test_frames.xyz. 
However, forces_PME is different. Yet the code that calculates forces_PME is used in README as a demonstration.

### PR DESCRIPTION
**Pull request by PicoCentauri**
Fixes #188 

The gradient recording was started before the tuning leading to wrong gradients when using `positions.grad`.

Thanks to @YuchiGuo for pointing this out.

# Contributor (creator of pull-request) checklist

 ~- [ ] Tests updated (for new features and bugfixes)?~
 - [x] Documentation updated (for new features)?
 - [x] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
📚 Documentation preview 📚: https://torch-pme--190.org.readthedocs.build/en/190/

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -100,6 +100,9 @@ Here is a simple example to get started with *torch-pme*:
    >>> potential = torchpme.CoulombPotential(smearing)
    >>> calculator = torchpme.P3MCalculator(potential, **p3m_parameters)
 
+   >>> # Start recording operations done to ``positions``
+   >>> _ = positions.requires_grad_()
+
```
Comment (E-Rum) [inline README.rst:105]: Do we actually need this dummy assignment, can't we just call it and ignore the return?

like: `positions.requires_grad_()`

Comment (PicoCentauri) [reply README.rst:105]: The function returns the array and if we don't use the dummy value our doctest will complain that there is a return value which we don't give.

Comment (E-Rum) [reply README.rst:105]: Sure, then it's fine, LGTM!

**Review by E-Rum — APPROVED** [review]
